src/sagemaker/serve/model_server/multi_model_server/server.py

```diff
@@ -0,0 +1,127 @@
+"""Module for the MultiModel Local and Remote servers"""
+from __future__ import absolute_import
+
+import requests
+import logging
+from pathlib import Path
+from sagemaker import Session, fw_utils
+from sagemaker.serve.utils.exceptions import LocalModelInvocationException
+from sagemaker.base_predictor import PredictorBase
+from sagemaker.s3_utils import determine_bucket_and_prefix, parse_s3_url, s3_path_join
+from sagemaker.s3 import S3Uploader
+from sagemaker.local.utils import get_docker_host
+
+MODE_DIR_BINDING = "/opt/ml/model/"
+_DEFAULT_ENV_VARS = {}
+
+logger = logging.getLogger(__name__)
+
+
+class LocalMultiModelServer:
+    """Local Multi Model server instance"""
+
+    def _start_serving(
+        self,
+        client: object,
+        image: str,
+        model_path: str,
+        secret_key: str,
+        env_vars: dict,
+    ):
+        """Placeholder docstring"""
+        self.container = client.containers.run(
+            image,
+            "serve",
+            network_mode="host",
+            detach=True,
+            auto_remove=True,
+            volumes={
+                Path(model_path).joinpath("code"): {
+                    "bind": MODE_DIR_BINDING,
+                    "mode": "rw",
+                },
+            },
+            environment=_update_env_vars(env_vars),
+        )
+
+    def _invoke_multi_model_server_serving(self, request: object, content_type: str, accept: str):
+        """Placeholder docstring"""
+        try:
+            response = requests.post(
+                f"http://{get_docker_host()}:8080/invocations",
+                data=request,
+                headers={"Content-Type": content_type, "Accept": accept},
+                timeout=600,
+            )
+            response.raise_for_status()
+            return response.content
+        except Exception as e:
+            raise Exception("Unable to send request to the local container server") from e
+
+    def _multi_model_server_deep_ping(self, predictor: PredictorBase):
+        """Placeholder docstring"""
+        response = None
+        try:
+            response = predictor.predict(self.schema_builder.sample_input)
+            return True, response
+            # pylint: disable=broad-except
+        except Exception as e:
+            if "422 Client Error: Unprocessable Entity for url" in str(e):
+                raise LocalModelInvocationException(str(e))
+            return False, response
+
+        return (True, response)
+
+
+class SageMakerMultiModelServer:
```

Comment:
There are a lot of duplication between MultiModelServer and TorchServer component code. Can we merge the duplicate code, so it is easy to maintain (like bug fix in one place).

Reply:
I can take this as a follow up if alright